src/middleware.ts

```diff
@@ -368,22 +368,23 @@ export const persist =
             )
           } else {
             stateFromStorageInSync = deserializedStorageValue.state
-            set(deserializedStorageValue.state)
           }
         }
       })
       .then((migratedState) => {
         if (migratedState) {
           stateFromStorageInSync = migratedState as S
-          set(migratedState as PartialState<S, keyof S>)
-          return setItem()
         }
-      })
-      .then(() => {
+
         stateFromStorageInSync = merge(
           stateFromStorageInSync as S,
           configResult
         )
+
+        set(stateFromStorageInSync as S)
+        return setItem()
+      })
+      .then(() => {
         postRehydrationCallback?.(stateFromStorageInSync, undefined)
```

Comment:
```suggestion
        postRehydrationCallback?.(stateFromStorage, undefined)
```
Now, as it' used for both sync/async, maybe rename it?

Reply:
good idea :+1: